dataset/non_huggingface_datasets.py

```diff
@@ -1,121 +1,92 @@
-import tempfile
-import zipfile
 import os
-import json
 import random
-import gdown
-from urllib.request import urlretrieve, urlopen
+import datasets
+from datasets import Dataset
+from tqdm import tqdm
 from typing import Optional, List, Tuple
 from dataset.st_dataset import SummInstance, SummDataset
 
 
+BASE_NONHUGGINGFACE_DATASETS_PATH = os.path.join(os.getcwd(), "dataset", "non_huggingface_datasets_builders")
```

Comment:
Thanks!

The `non_huggingface_datasets.py` and the  non_huggingface DatasetBuilderClasses are indeed related and this is inspired/necessitated by the huggingface pipeline. The `BuilderClasses` contain the script to download the datasets. Although Huggingface appears to host the datasets, they do not store any of the data. Instead, they have the dataset creator submit a script subclassing their own `DatasetBuilderClass` that downloads the data from wherever it is hosted. 

I am leveraging this functionality by creating local scripts for all the non-huggingface datasets that we support and letting the huggingface API handle all the overhead for downloading the corpora. A big benefit and motivation for this is that it includes the handles caching system as well as having other advantages such as robust and simplified downloads. So the builder scripts are used to download the datasets and then prepackage them as if they were any other huggingface dataset.

Once the data is downloaded, that's where the `non_huggingface_datasets.py` classes come into play. Similar to how the `huggingface_datasets.py` is used to post-process the data into our own dataset class, `non_huggingface_datasets.py` does the same. Due to adopting the huggingface pipeline, there's very little difference between these two files now other than their arguments to the `load_dataset` method. (Perhaps, we should merge the two). So the BuilderClasses are used for downloading the dataset while the `huggingface_datasets.py and `non_huggingface_datasets.py` are used to post-process the data.


The choice for using the path instead of `import` is a great question and one that has generated a lot of consideration. Similar to the first issue, this choice is due to the huggingface pipeline.
So how this works is the huggingface `dataset.load_dataset` method takes the location of the dataset builder script and then uses the DatasetBuilderClass contained in it to download the data or fetch from cache. Unfortunately as it is, there's no other way of passing in the contents of the builder script to the `load_dataset' method other` than providing the path to it. The only way to bypass this is to create our own `load_dataset` but this diverges from the pipeline as we discussed last week (and also loses a few features). This was the issue I was referring to.  So I used the huggingface way and passing in the `dataset builder` paths. This has turned out to be ideal since we are able to leverage the full utility of the huggingface API and end up solving other issues caused by other third party modules such as `gdown` that have now been eliminated creating a more robust pipeline.

Reply:
Thanks a lot for the thorough explanation! 

I see, so much of these seemingly " suboptimal " choices are actually necessary to accommodate huggingface's API. However, maybe you can checkout [this page](https://stackoverflow.com/questions/3430372/how-do-i-get-the-full-path-of-the-current-files-directory/54892023) to see an alternative for `os.getcwd()`?